versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -825,6 +827,120 @@ public final CommitLogEntry fetchFromCommitLog(OP_CONTEXT ctx, Hash hash) {
 
   protected abstract CommitLogEntry doFetchFromCommitLog(OP_CONTEXT ctx, Hash hash);
 
+  @Override
+  public Stream<CommitLogEntry> scanAllCommitLogEntries() {
+    OP_CONTEXT ctx = borrowConnection();
+    return doScanAllCommitLogEntries(ctx)
+        .onClose(
+            () -> {
+              try {
+                ctx.close();
+              } catch (Exception e) {
+                throw new RuntimeException(e);
+              }
+            });
+  }
+
+  @Override
+  public HeadsAndForkPoints identifyAllHeadsAndForkPoints(int expectedCommitCount) {
```

Comment:
This functionality appears to be kind of external to `DatabaseAdapter`. Would it make sense to move this code to a utility class?.. or maybe a separate CDI bean? Do we expect to have adapter-specitic logic in this method the future? :thinking:

Reply:
moved out